java/compression/src/test/java/org/apache/arrow/compression/TestCompressionCodecFile.java

```diff
@@ -0,0 +1,176 @@
+import org.apache.arrow.vector.compression.NoCompressionCodec;
+import org.apache.arrow.vector.ipc.ArrowFileReader;
+import org.apache.arrow.vector.ipc.ArrowFileWriter;
+import org.apache.arrow.vector.ipc.ArrowReader;
+import org.apache.arrow.vector.ipc.ArrowStreamReader;
+import org.apache.arrow.vector.ipc.ArrowStreamWriter;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class TestCompressionCodecFile {
+  private final CompressionCodec codec;
+  private final int vectorLength;
+  private final File file;
+
+  public TestCompressionCodecFile(CompressionUtil.CodecType type, int vectorLength, CompressionCodec codec,
+                                  File file) {
+    this.codec = codec;
+    this.file = file;
+    this.vectorLength = vectorLength;
+  }
+
+  @Parameterized.Parameters(name = "codec = {0}, length = {1}, file = {2}")
+  public static Collection<Object[]> getCodecs() {
+    List<Object[]> params = new ArrayList<>();
+
+    int[] lengths = new int[]{10, 100, 1000};
+    for (int len : lengths) {
+      CompressionCodec dumbCodec = NoCompressionCodec.INSTANCE;
+      File fileNoCompression = new File("target/write_no_compression_" + len + ".arrow");
+      params.add(new Object[]{dumbCodec.getCodecType(), len, dumbCodec, fileNoCompression});
+
+      CompressionCodec lz4Codec = new Lz4CompressionCodec();
+      File fileLZ4Compression = new File("target/write_lz4_compression_" + len + ".arrow");
+      params.add(new Object[]{lz4Codec.getCodecType(), len, lz4Codec, fileLZ4Compression});
+
+      CompressionCodec zstdCodec = new ZstdCompressionCodec();
+      File fileZSTDCompression = new File("target/write_zstd_compression_" + len + ".arrow");
+      params.add(new Object[]{zstdCodec.getCodecType(), len, zstdCodec, fileZSTDCompression});
+
+    }
+    return params;
+  }
+
+  @Test
+  public void writeReadRandomAccessFile() throws IOException {
+    RootAllocator allocator = new RootAllocator(Long.MAX_VALUE);
+    BitVector bitVector = new BitVector("boolean", allocator);
+    VarCharVector varCharVector = new VarCharVector("varchar", allocator);
+    for (int i = 0; i < vectorLength; i++) {
+      bitVector.setSafe(i, i % 2 == 0 ? 0 : 1);
+      varCharVector.setSafe(i, ("test" + i).getBytes(StandardCharsets.UTF_8));
+    }
+    bitVector.setValueCount(vectorLength);
+    varCharVector.setValueCount(vectorLength);
+
+    List<Field> fields = Arrays.asList(bitVector.getField(), varCharVector.getField());
+    List<FieldVector> vectors = Arrays.asList(bitVector, varCharVector);
+
+    VectorSchemaRoot schemaRootWrite = new VectorSchemaRoot(fields, vectors);
+
+    // write
+    FileOutputStream fileOutputStream = new FileOutputStream(file);
+    ArrowFileWriter writer = new ArrowFileWriter(schemaRootWrite, true, codec, true,
```

Comment:
please try to comment literal paraemters.  e.g. `schemarRootWrite, /*parameter 1 name=*/ true, /*parameter 2 name=*/`